plugins/source/aws/client/account.go

```diff
@@ -100,17 +101,25 @@ func findEnabledRegions(ctx context.Context, logger zerolog.Logger, accountName
 		regionsToCheck = localRegions
 	}
 
+	deniedErrors := []string{}
 	for _, region := range regionsToCheck {
 		enabledRegions, err := getEnabledRegions(ctx, ec2Client, region)
 		if err != nil {
 			logger.Warn().Str("account", accountName).Err(err).Msgf("Failed to find disabled regions for account when checking: %s", region)
+			if IsAWSError(err, "UnauthorizedOperation") {
+				deniedErrors = append(deniedErrors, region)
+			}
 			continue
 		}
 		filteredRegions := filterDisabledRegions(localRegions, enabledRegions)
 		if len(filteredRegions) > 0 {
 			return filteredRegions
 		}
 	}
+	if !isAllRegions(localRegions) && len(deniedErrors) == len(regionsToCheck) && funk.Subset(regionsToCheck, deniedErrors) {
+		logger.Info().Str("account", accountName).Msg("Access denied for all regions, assuming all specified regions are enabled")
+		return localRegions
+	}
```

Comment:
I suggest adding a warning here add well, to indicate that no reasons are left (nit)